client/data/gutenberg/fse-settings-query.ts

```diff
@@ -0,0 +1,43 @@
+import { useLayoutEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import { getStoredItem, setStoredItem } from 'calypso/lib/browser-storage';
+import wpcom from 'calypso/lib/wp';
+
+export type GutenbergFSESettings = {
+	is_core_fse_eligible: boolean;
+};
+
+export const useGutenbergFSESettingsQuery = ( siteId: string ) => {
```

Comment:
@jsnajdr let me know if this is something we want to scale/make more generic. I don't think we have a caching strategy set for `react-query` but I think this is convenient enough to replicate in other places.

Reply:
If you mean storing the query cache to local storage, I implemented a generic solution today in #56828. Instead of caching individual queries, it saves and restores the entire query cache using React Query's own experimental plugin.

Once that's landed, you can just `useQuery( key, () => wpcom.req.get( ... ) )` in your application code and never worry about local storage and caching again. It's happening automatically.

Reply:
Your solution works great! I just tested it. Excited for it to get merged to `trunk` so I can get a copy on my branch!